packages/react-ui-components/src/Headline/headline.tsx

```diff
@@ -0,0 +1,76 @@
+import mergeClassNames from 'classnames';
+import React, {PureComponent} from 'react';
+
+import {PickDefaultProps} from '../../types';
+
+type HeadlineType = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
+
+interface HeadlineTheme {
+    readonly 'heading': string;
+    readonly 'heading--h1': string;
+}
+
+interface HeadlineProps {
+    /**
+     * The contents to be rendered.
+     */
+    readonly children: React.ReactNode;
+
+    /**
+     * The semantic tag type of the headline.
+     */
+    readonly type: HeadlineType;
+
+    /**
+     * Optional style identifier, this enables the possibility to diff the
+     * semantic value of the UI to the displayed style.
+     */
+    readonly style?: HeadlineType;
```

Comment:
What is this `style` property supposed to do? We can't just map it to the style attribute of `h1` for example.

I can understand it's intention from the docs above, but it's just not what the code does.

Reply:
I can only think of type is for the tag i.e. h1, h2 etc. and style for the visual representation i.e. a h2-tag which looks like an h3 or something.
But that's just a guess.

Reply:
I can only think of type is for the tag i.e. h1, h2 etc. and style for the visual representation i.e. a h2-tag which looks like an h3 or something.
But that's just a guess - I think it's a leftover which nobody cleaned up.

Reply:
@JamesAlias fully ack, I checked the code as well; I think it's a leftover which is not in use right now - so we can get rid of the `style` attribute.

Reply:
As the PR was merged before I had time to remove it, this will come in a separate PR.